<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SconeStudioClass</class>
 <widget class="QMainWindow" name="SconeStudioClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1077</width>
    <height>802</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SCONE</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>4</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="currentIndex">
       <number>-1</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideMiddle</enum>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1077</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="resultsDock">
   <property name="minimumSize">
    <size>
     <width>160</width>
     <height>101</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>640</width>
     <height>524287</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Optimization Results</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="resultsDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <property name="spacing">
      <number>4</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="QFileBrowser" name="resultsBrowser">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <attribute name="headerVisible">
        <bool>true</bool>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="messagesDock">
   <property name="minimumSize">
    <size>
     <width>91</width>
     <height>101</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Messages</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="messagesDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="spacing">
      <number>4</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="QLogSink" name="outputText">
       <property name="font">
        <font>
         <family>Consolas</family>
         <pointsize>9</pointsize>
        </font>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="viewerDock">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Viewer</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="viewerDockContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>4</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QOsgViewer" name="osgViewer" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>322</width>
         <height>322</height>
        </size>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QStackedWidget" name="stackedWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="playPage">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>4</number>
         </property>
         <property name="leftMargin">
          <number>2</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QPlayControl" name="playControl" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>24</height>
            </size>
           </property>
           <property name="autoFillBackground">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="progressPage">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>4</number>
         </property>
         <property name="leftMargin">
          <number>2</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QProgressBar" name="progressBar">
           <property name="value">
            <number>24</number>
           </property>
           <property name="textVisible">
            <bool>true</bool>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="textDirection">
            <enum>QProgressBar::TopToBottom</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="abortButton">
           <property name="text">
            <string>Abort</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="4" margin="4"/>
 <customwidgets>
  <customwidget>
   <class>QOsgViewer</class>
   <extends>QWidget</extends>
   <header>qosgviewer.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QPlayControl</class>
   <extends>QWidget</extends>
   <header>qplaycontrol.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QFileBrowser</class>
   <extends>QTreeView</extends>
   <header>QFileBrowser.h</header>
  </customwidget>
  <customwidget>
   <class>QLogSink</class>
   <extends>QPlainTextEdit</extends>
   <header>qlogsink.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="SconeStudio.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>resultsBrowser</sender>
   <signal>activated(QModelIndex)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>activateBrowserItem(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>620</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>tabCloseRequested(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>636</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resultsDock</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fixViewCheckboxes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>messagesDock</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fixViewCheckboxes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>673</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>activateBrowserItem(QModelIndex)</slot>
  <slot>updateScrollbar(int)</slot>
  <slot>start()</slot>
  <slot>stop()</slot>
  <slot>slomo(int)</slot>
  <slot>fileOpen()</slot>
  <slot>fileClose()</slot>
  <slot>fileSave()</slot>
  <slot>fileSaveAs()</slot>
  <slot>showViewer()</slot>
  <slot>viewAction()</slot>
  <slot>helpAbout()</slot>
  <slot>optimizeScenario()</slot>
  <slot>abortOptimizations()</slot>
  <slot>fileExit()</slot>
  <slot>selectBrowserItem(QModelIndex)</slot>
  <slot>createVideo()</slot>
  <slot>optimizeScenarioMultiple()</slot>
  <slot>tabCloseRequested(int)</slot>
  <slot>updateViewSettings()</slot>
  <slot>showSettingsDialog()</slot>
  <slot>viewResults(bool)</slot>
  <slot>viewMessages(bool)</slot>
  <slot>fixViewCheckboxes()</slot>
  <slot>runScenario()</slot>
 </slots>
</ui>
